nextcord/command_client.py

```diff
@@ -0,0 +1,748 @@
+            return result
+        return decorator
+
+
+class ApplicationCommand(ApplicationSubcommand):
+    def __init__(self, callback: Callable, cmd_type: CommandType,
+                 name: str = "", description: str = "", guild_ids: List[int] = None, force_global: bool = False,
+                 default_permission: Optional[bool] = None):
+        # TODO: Have global and guilds be off. Allow a command to be in the system, but not registered to guilds or
+        #  global. Global should be True, False, None where None makes it default to True if no guild_ids and false
+        #  if there are guild_ids. For super dynamic guild_id setting, it will be done at runtime. Thus, being able to
+        #  have a command not be global and not be a guild command must be possible. Index by type + name.
+        super().__init__(callback=callback, parent=None, cmd_type=cmd_type, name=name, description=description,
+                         guild_ids=None)
+        # Basic input checking.
+        if guild_ids is None:
+            guild_ids = []
+        self._state: Optional[ConnectionState] = None  # TODO: I thought there was a way around doing this, but *sigh*.
+        self._is_global: Optional[bool] = True if (guild_ids and force_global) or (not guild_ids) else False
+        self._is_guild: Optional[bool] = True if guild_ids else False
+
+        self.default_permission: Optional[bool] = default_permission
+        self.guild_ids: List[int] = guild_ids
+        self.type = cmd_type
+        self._global_id: Optional[int] = None
+        self._guild_ids: Dict[int, int] = {}  # Guild ID is key, command ID is value.
+
+    def parse_response(self, response: ApplicationCommandResponse):
+        self.raw_parse_result(response._state, response.guild_id, response.id)
+
+    def raw_parse_result(self, state: ConnectionState, guild_id: Optional[int], command_id):
+        self._state = state
+        if guild_id:
+            self._guild_ids[guild_id] = command_id
+        else:
+            self._global_id = command_id
+
+    async def call_from_interaction(self, interaction: Interaction):
+        """Runs call using the held ConnectionState object and given interaction."""
+        if not self._state:
+            raise NotImplementedError("State hasn't been set yet, this isn't handled yet!")
+        await self.call(self._state, interaction, interaction.data.get("options", {}))
+
+    async def call(self, state: ConnectionState, interaction: Interaction, option_data: List[Dict[str, Any]]):
+        """
+        Calls the callback, gathering and inserting kwargs into the callback as needed.
+        This handles CommandTypes that subcommands cannot handle, such as Message or User commands.
+        :param state: ConnectionState to fetch objects from cache.
+        :param interaction: Current Interaction object.
+        :param option_data: List of options, typically 'options' in the interaction data payload.
+        """
```

Comment:
```suggestion
        """|coro|

        Calls the callback, gathering and inserting kwargs into the callback as needed.
        This handles CommandTypes that subcommands cannot handle, such as Message or User commands.

        state: :class:`ConnectionState`
            ConnectionState to fetch objects from cache.
        interaction: :class:`Interaction`
            Current Interaction object.
        option_data: List[Dict[:class:`str`, Any]]
            List of options, typically 'options' in the interaction data payload.
        """
```
probably one of your old docstr's